### PR TITLE
Auth UI

### PR DESCRIPTION
**Pull request by muskanbagrecha**
1. Added auth functionality using firebase.
2. Users can login through email, password or through Google sign-in.
3. Users can signup
4. Ignore commented code as of now, it will be handled in successive PRs. 
VIEW ONLY: [files](https://github.com/muskanbagrecha/social-media-react-app/pull/2/files/fd901e8ed319df3cd30de844cc0ae8c20aecef83..0fb34f063b4558017af0482246e8f916b185d0ec)
Preview Link : https://deploy-preview-2--fashgram-media-react.netlify.app/

### CODE REVIEW

**Review by SinchanaMS — APPROVED** [review]
Couldn't find any issues. 
Liked how modular and segregated everything is. Good work!!

File: src/pages/signup/Signuppage.tsx [hunk 1]
```diff
@@ -1,3 +1,159 @@
...
+            <form onSubmit={(e) => e.preventDefault()}>
+              <div className="mb-6">
+                <input
+                  type="text"
+                  className="form-control block w-full px-4 py-2 text-l font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-primary-500 focus:outline-none"
+                  placeholder="John Doe"
+                  onChange={signupInputChangeHandler}
+                  name="name"
+                  required
+                />
```
Comment (zoef-yt) [inline src/pages/signup/Signuppage.tsx:73]: You can make this into one component and reuse it.

File: src/pages/profilepage/components/main-profile/MainProfile.tsx [hunk 2]
```diff
@@ -1,33 +1,55 @@
...
+          <Link to="/" className="text-primary-500 text-sm">
+            muskanbagrecha04@gmail.com
+          </Link>
+        </div>
+      </div>
+      <div className="flex text-sm gap-8">
+        <div className="flex flex-col items-center">
+          <span>10</span>
+          <span>Followers</span>
+        </div>
```
Comment (zoef-yt) [inline src/pages/profilepage/components/main-profile/MainProfile.tsx:38]: Make this into a component and send props.

File: src/pages/homepage/components/who-to-follow/WhoToFollow.tsx [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
-              </div>
+            <div className="flex items-center px-2 w-full rounded hover:cursor-pointer hover:bg-gray-100 dark:hover:bg-gray-700">
+              <img src={user.avatar} alt="user" className="avatar avatar-xs" />
+              <p className="text-sm text-gray-200">{user.name}</p>
+              <button className="text-base text-primary-500 hover:text-primary-200 ml-auto">
+                Follow
+              </button>
+              {/* <button className="text-base text-gray-500 hover:text-primary-200 ml-auto">
+                Unfollow
+              </button> */}
```
Comment (zoef-yt) [inline src/pages/homepage/components/who-to-follow/WhoToFollow.tsx:22]: remove commented code.

File: src/pages/homepage/components/main-feed/MainFeed.tsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+// import { Moments } from "../moments/Moments";
 import { CreatePost } from "../create-post/CreatePost";
 import { Post } from "../post/Post";
 import "./MainFeed.css";
 
 export const MainFeed: React.FC = () => {
   return (
     <main className="w-full md:w-3/5 lg:w-3/5 flex flex-col main-feed rounded ">
-      <Moments />
+      {/* <Moments /> */}
```
Comment (zoef-yt) [inline src/pages/homepage/components/main-feed/MainFeed.tsx:9]: remove commented code.

File: src/pages/homepage/components/main-feed/MainFeed.tsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
-import { Moments } from "../moments/Moments";
+// import { Moments } from "../moments/Moments";
```
Comment (zoef-yt) [inline src/pages/homepage/components/main-feed/MainFeed.tsx:1]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            following: 0,
+            posts: 0,
+            bio: '',
+            portfolio: '',
+            backgroundImageURL: '',});
+        // dispatch(setIsLoading(false));
+    } catch (error) {
+        console.log(error);
+        // errorHandler(true, error.message);
+        // dispatch(setIsLoading(false));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:81]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            uid: auth.currentUser?.uid,
+            photoURL: auth.currentUser?.photoURL,
+            email: auth.currentUser?.email,
+            followers: 0,
+            following: 0,
+            posts: 0,
+            bio: '',
+            portfolio: '',
+            backgroundImageURL: '',});
+        // dispatch(setIsLoading(false));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:77]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+	}
+}
+
+export const signupHandler = async(name: string, email : string, password : string, dispatch: Function) => {
+    try {
+        // dispatch(setIsLoading(true));
+        const userCredential = await createUserWithEmailAndPassword(auth, email, password);
+        await updateProfile(userCredential.user, {
+			displayName: name,
+			// photoURL: defaultImage,
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:64]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+	} catch (error : any) {
+		console.log(error.message);
+		// errorHandler(true, error.message);
+		// dispatch(setIsLoading(false));
+	}
+}
+
+export const signupHandler = async(name: string, email : string, password : string, dispatch: Function) => {
+    try {
+        // dispatch(setIsLoading(true));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:60]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+		await signInWithEmailAndPassword(auth, email, password);
+        console.log(auth.currentUser);
+        
+        dispatch(setAuthUser({uid: auth?.currentUser?.uid, email: "", displayName: "", photoURL: "", bio: "", portfolio: "", backgroundImageURL: "", followers: 0, following: 0, posts: 0}));
+        
+		// dispatch(setIsLoading(false));
+	} catch (error : any) {
+		console.log(error.message);
+		// errorHandler(true, error.message);
+		// dispatch(setIsLoading(false));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:54]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+export const loginHandler = async(email : string, password : string, dispatch: Function) => {
+    try {
+		// dispatch(setIsLoading(true));
+		await signInWithEmailAndPassword(auth, email, password);
+        console.log(auth.currentUser);
+        
+        dispatch(setAuthUser({uid: auth?.currentUser?.uid, email: "", displayName: "", photoURL: "", bio: "", portfolio: "", backgroundImageURL: "", followers: 0, following: 0, posts: 0}));
+        
+		// dispatch(setIsLoading(false));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:50]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+    } catch (error) {
+        console.log(error);
+        // errorHandler(true, error.message);
+        // dispatch(setIsLoading(false));
+    }
+};
+
+export const loginHandler = async(email : string, password : string, dispatch: Function) => {
+    try {
+		// dispatch(setIsLoading(true));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:44]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+            following: 0,
+            posts: 0,
+            bio: '',
+            portfolio: '',
+            backgroundImageURL: '',});
+        // dispatch(setIsLoading(false));
+    } catch (error) {
+        console.log(error);
+        // errorHandler(true, error.message);
+        // dispatch(setIsLoading(false));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:38]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+            uid: auth.currentUser?.uid,
+            photoURL: auth.currentUser?.photoURL,
+            email: auth.currentUser?.email,
+            followers: 0,
+            following: 0,
+            posts: 0,
+            bio: '',
+            portfolio: '',
+            backgroundImageURL: '',});
+        // dispatch(setIsLoading(false));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:34]: remove commented code.

File: src/firebase/firebaseAuth.ts [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+import {setAuthUser} from "../store/auth-action/authSlice";
+import { addUserToTheDB, getUser } from "./firebase-firestore";
+
+export const auth = getAuth(app);
+
+const provider = new GoogleAuthProvider();
+
+export const googleAuthHandler = async (dispatch : Function) => {
+    try {
+        // dispatch(setIsLoading(true));
```
Comment (zoef-yt) [inline src/firebase/firebaseAuth.ts:20]: remove commented code.

**Review by zoef-yt — APPROVED** [review]
The UI looks really good nice to see you are using TS. Great work Muskan good to merge.

**Comment by muskanbagrecha** [comment]
All Suggestions have been implemented in next PRs.

**Pull request merged**